File: bounded-contexts/java/src/main/java/p1/myshop/services/shipping/DeliveryService.java
package p1.myshop.services.shipping;

import p1.myshop.entities.HasWeight;
import p1.myshop.services.catalog.CatalogService;
import p1.myshop.services.shoppingcart.Cart;
import p1.myshop.services.shoppingcart.CartItem;

public class DeliveryService {
    private final CatalogService catalogService;

    public DeliveryService(CatalogService catalogService) {
        this.catalogService = catalogService;
    }

    public double calculateOrderWeight(Cart cart) {
        return cart.items().stream()
                .map(CartItem::itemId)
                .map(catalogService::loadItem)
                .mapToDouble(HasWeight::weight)
                .sum();
    }
}

File: bounded-contexts/dotnet/MyShop.Services/Shipping/DeliveryService.cs
using MyShop.Services.ShoppingCart;
using System;
using System.Linq;
using System.Collections.Generic;
using System.Text;

namespace MyShop.Services.Shipping
{
    public class DeliveryService
    {
        public DeliveryService(CatalogService catalogService)
        {
            CatalogService = catalogService;
        }

        public CatalogService CatalogService { get; }

        public double CalculateOrderWeight(Cart cart)
        {
            return cart.GetItems()
                .Sum(item => item.Quantity * CatalogService.LoadItem(item.ItemId).Weight);
        }
    }
}
